### PR TITLE
bpo-27115: Use Query subclass for IDLE editor Goto

### PR DESCRIPTION
**Pull request by terryjreedy**
Replace tkinter tkSimpleDialog.askinteger with IDLE query dialog.
The new box checks for positivity before returning.

https://bugs.python.org/issue27115

### CODE REVIEW

File: Lib/idlelib/query.py [hunk 1]
```diff
@@ -224,6 +224,25 @@ def entry_ok(self):
         return file_path
 
 
+class Goto(Query):
+    "Get a positive line number for editor Go To Line."
+    # Used in editor.EditorWindow.goto_line_event.
+
+    def __init__(self, parent, title, message, *, _htest=False, _utest=False):
```
Comment (taleinat) [inline Lib/idlelib/query.py]: This `__init__` seems unnecessary.

File: Lib/idlelib/editor.py [hunk 2]
```diff
@@ -672,14 +672,11 @@ def replace_event(self, event):
...
         text = self.text
-        lineno = tkSimpleDialog.askinteger("Goto",
-                "Go to line number:",parent=text)
-        if lineno is None:
-            return "break"
-        if lineno <= 0:
-            text.bell()
-            return "break"
-
+        lineno = query.Goto(
```
Comment (taleinat) [inline Lib/idlelib/editor.py:675]: Does this behave correctly when simply closing the dialog?

File: Lib/idlelib/NEWS.txt [hunk 3]
```diff
@@ -3,6 +3,9 @@ Released on 2020-10-05?
 ======================================
 
 
+bpo-27115: For Edit Go to Line, use a Query entry box subclass with
```
Comment (taleinat) [inline Lib/idlelib/NEWS.txt]: "Edit Go to Line" is a bit awkward; IMO "Go to Line" is sufficient and clearer.

(Relevant to the NEWS entry as well.)

File: Lib/idlelib/query.py (hunk 1 above)
Comment (terryjreedy) [reply Lib/idlelib/query.py]: Right, I copied and edited from an existing subclass that does need it.  Deleted.

File: Lib/idlelib/editor.py (hunk 2 above)
Comment (terryjreedy) [reply Lib/idlelib/editor.py:675]: No. Cursor stays put but selection is erroneously cleared.  Fixed.

File: Lib/idlelib/NEWS.txt (hunk 3 above)
Comment (terryjreedy) [reply Lib/idlelib/NEWS.txt]: Changed.  Thank you for the comments.

**Pull request merged**